File: RoomBasic3/app/src/main/java/com/example/roombasic/MyAdapter.java
package com.example.roombasic;


import android.content.Intent;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.MyViewHodler> {

    List<Word> allWords = new ArrayList<>();
    boolean useCardView;

    public MyAdapter(boolean useCardView) {
        this.useCardView = useCardView;
    }

    public void setAllWords(List<Word> allWords) {
        this.allWords = allWords;
    }



    @NonNull
    @Override
    public MyViewHodler onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        View itemView = layoutInflater.inflate(useCardView ?  R.layout.cell_card : R.layout.cell_nomal, parent, false);
        return new MyViewHodler(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHodler holder, int position) {
        Word word = allWords.get(position);
//        holder.textViewNumber.setText(String.valueOf(word.getId()));
        holder.textViewNumber.setText(String.valueOf(position)); // 显示序号
        holder.textViewEnglish.setText(word.getWord());
        holder.textViewChinese.setText(word.getChineseMeaning());
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Uri uri = Uri.parse("https://m.youdao.com/dict?le=eng&q="+ holder.textViewEnglish.getText());
                Intent intent = new Intent(Intent.ACTION_VIEW);
                intent.setData(uri);
                holder.itemView.getContext().startActivity(intent);
            }
        });

    }

    @Override
    public int getItemCount() {
        return allWords.size();
    }

    static class MyViewHodler extends RecyclerView.ViewHolder {
        TextView textViewNumber, textViewEnglish, textViewChinese;

        public MyViewHodler(@NonNull View itemView) {
            super(itemView);

            textViewNumber = itemView.findViewById(R.id.textViewNumber);
            textViewEnglish = itemView.findViewById(R.id.textViewEnglish);
            textViewChinese = itemView.findViewById(R.id.textViewChinese);
        }
    }
}

File: LeanDemo/app/src/main/java/com/example/leandemo/MyAdapter.kt
package com.example.leandemo

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import cn.leancloud.LCObject

class MyAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    private var _dataList = listOf<LCObject>()

    fun updateDataList(newList : List<LCObject>) {
        _dataList = newList
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.view_holder, parent, false)
        return object : RecyclerView.ViewHolder(v){}
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        val ob = _dataList[position]
        holder.itemView.findViewById<TextView>(R.id.textViewContent).text = ob.get("word").toString()
        holder.itemView.findViewById<TextView>(R.id.textViewTime).text = ob.get("createdAt").toString()

    }

    override fun getItemCount(): Int {
        return _dataList.size
    }
}